### PR TITLE
Check if client outdated

### LINKED ISSUE #916: Inform user in the app if she is running on an outdated version
It happened that a user was running on an outdated version. It might help if the app tells the user directly that a new version is available.

We could event go so far to disable some functionalities if the app is not on the latest version.

### PR DESCRIPTION
**Pull request by Restioson**
This PR is based on the advice from @klochowicz in #916. It introduces a new `/api/version` endpoint to the coordinator, which returns its version as recorded in `coordinator/Cargo.toml`. This, in turn, is kept in sync by the `draft_new_release` action (I am not quite sure how to test it locally). Then, on startup, the mobile client will perform a GET request to `/api/version`, compare it to its own version, and if it is out of date it will create a snackbar notification.

I'm not sure if we want to add an action to direct the user to an update page, or to add something else to the snackbar.

There might also be considerations for local/regtest development where we might want to handle this differently.

In line with what was discussed, this PR does not disable anything, merely informing the user that they are out of date. A future PR could introduce a confirm prompt for taking actions (or on startup) if the user is deemed "too" out of date such that it would be dangerous.

Fixes #916.

### CODE REVIEW

File: mobile/lib/main.dart [hunk 1]
```diff
@@ -221,7 +228,34 @@ class _TenTenOneAppState extends State<TenTenOneApp> {
...
+      try {
+        final response = await http.get(
+          Uri.parse('http://${config.host}:${config.httpPort}/api/version'),
+        );
+
+        final clientVersion = Version.parse(packageInfo.version);
+        final coordinatorVersion = Version.parse(jsonDecode(response.body));
+        FLog.info(text: "Coordinator version: ${coordinatorVersion.toString()}");
+
+        if (coordinatorVersion > clientVersion) {
```
Comment (holzeis) [inline mobile/lib/main.dart:249]: 🤔 will that work for something like "1.0.21" and "1.1.0" ?

File: mobile/lib/main.dart [hunk 2]
```diff
@@ -221,7 +228,34 @@ class _TenTenOneAppState extends State<TenTenOneApp> {
...
+          messenger.showSnackBar(SnackBar(
```
Comment (holzeis) [inline mobile/lib/main.dart:250]: 🔧 I think a snack bar is not the most prominent way of informing the user that she is running an outdated version. How about a popup, that the user has to confirm?

**Review by holzeis — COMMENTED** [review]
Can you add a screenshot? I am not sure if a snackbar is sufficient to inform the user about an outdated version.

File: mobile/lib/main.dart (hunk 1 above)
Comment (Restioson) [reply mobile/lib/main.dart:249]: This is using a SemVer parsing library, and I believe 1.0.21 should compare as less than 1.1.0 in SemVer

File: mobile/lib/main.dart (hunk 2 above)
Comment (holzeis) [reply mobile/lib/main.dart:250]: I would argue for it, but we don't have to do it in this change.

Comment (Restioson) [reply mobile/lib/main.dart:250]: I'm happy to circle back to this tomorrow and get it in!

Comment (Restioson) [reply mobile/lib/main.dart]: It now shows an alert dialogue

**Comment by Restioson** [comment]
image

**Comment by holzeis** [comment]
Looks like there is unnecessary space in front of the second paragraph.

File: mobile/lib/main.dart [hunk 3]
```diff
@@ -221,7 +228,48 @@ class _TenTenOneAppState extends State<TenTenOneApp> {
...
+
+        if (coordinatorVersion > clientVersion) {
+          FLog.warning(text: "Client out of date. Current version: ${clientVersion.toString()}");
+          showDialog(
+              context: _shellNavigatorKey.currentContext!,
+              builder: (context) => AlertDialog(
+                      title: const Text("Update available"),
+                      content: Text("A new version of 10101 is available: "
+                          "${coordinatorVersion.toString()}.\n\n"
+                          " Please note that if you do not update 10101, the app"
```
Comment (holzeis) [inline mobile/lib/main.dart:257]:
```suggestion
                          "Please note that if you do not update 10101, the app"
```

**Review by holzeis — APPROVED** [review]

**Comment by Restioson** [comment]
I'll merge this shortly, accidentally excluded pubspec.yml

**Pull request merged**